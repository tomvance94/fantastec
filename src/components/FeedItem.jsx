import React from 'react'

function buildClassString (id, activeKeyMomentId) {
  let classes = ['LiveFeed__item', 'cf']
  if (id === activeKeyMomentId) classes.push('active')

  return classes.join(' ')
}

export default ({
  id,
  interval,
  activeKeyMomentId,
  comment
}) => (
  <li className={buildClassString(id, activeKeyMomentId)}>
    <span>{interval}</span>
    <p>{comment}</p>
  </li>
)
