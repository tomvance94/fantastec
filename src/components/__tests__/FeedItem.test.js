import React from 'react';
import FeedItem from '../FeedItem';
import renderer from 'react-test-renderer';

test('When active it has the active class', () => {
  const component = renderer.create(
    <FeedItem 
      id={1}
      interval='10:00'
      activeKeyMomentId={0}
      comment='The Comment String'
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.activeKeyMomentId = 1;
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});