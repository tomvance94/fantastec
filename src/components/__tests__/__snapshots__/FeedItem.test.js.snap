// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When active it has the active class 1`] = `
<li
  className="LiveFeed__item cf"
>
  <span>
    10:00
  </span>
  <p>
    The Comment String
  </p>
</li>
`;

exports[`When active it has the active class 2`] = `
<li
  className="LiveFeed__item cf"
>
  <span>
    10:00
  </span>
  <p>
    The Comment String
  </p>
</li>
`;
